otherlibs/unix/stat_win32.c

```diff
@@ -339,50 +339,54 @@ static int do_stat(int do_lstat, int use_64, const char* opath, HANDLE fstat, __
 
 CAMLprim value caml_unix_stat(value path)
 {
+  CAMLparam1(path);
   struct _stat64 buf;
   __int64 st_ino;
 
   caml_unix_check_path(path, "stat");
   if (!do_stat(0, 0, String_val(path), NULL, &st_ino, &buf)) {
     caml_uerror("stat", path);
```

Comment:
Just to elaborate on @mlasson's explanation, the problem is this call to `caml_uerror`: if the GC runs during the execution of `do_stat` (which may happen if there are multiple threads), then `path` may no left dangling as it is not registered with the GC.